src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java

```diff
@@ -160,20 +165,39 @@ protected FileMonitoringController(FilePath ws) throws IOException, InterruptedE
             }
         }
 
-        // TODO would be more efficient to allow API to consolidate writeLog with exitStatus (save an RPC call)
-        @Override public Integer exitStatus(FilePath workspace, Launcher launcher, TaskListener listener) throws IOException, InterruptedException {
-            FilePath status = getResultFile(workspace);
-            if (status.exists()) {
-                try {
-                    return Integer.parseInt(status.readToString().trim());
-                } catch (NumberFormatException x) {
-                    throw new IOException("corrupted content in " + status + ": " + x, x);
+        /** Avoids excess round-tripping when reading status file. */
+        static class StatusCheck extends MasterToSlaveFileCallable<Integer> {
+            @Override
+            @CheckForNull
+            public Integer invoke(File f, VirtualChannel channel) throws IOException, InterruptedException {
+                if (f.exists() && f.length() > 0) {
+                    try {
+                        String fileString = Files.readFirstLine(f, Charset.defaultCharset());
```

Comment:
Anyway you could just get rid of the `exists` check and use a single `readToString` call, catching `FileNotFoundException` and ignoring.

Reply:
If it's all agent-local then it doesn't really matter, does it?  Also: as a rule of thumb in Java one should not be throwing Exceptions to signal routine and expected conditions.  They come with a higher than normal level of baggage. 

As opposed to Python where that's the idiomatic way to signal things like an Iterator being done.

Reply:
Well, Remoting itself is a pretty high level of baggage, which would swamp any tiny overhead of `Throwable.fillInStackTrace` in this case.